backstory/src/main/java/com/google/sps/servlets/data/BlobstoreManager.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets.data;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.sps.servlets.data.BlobstoreServiceConstantFields;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Wrapper class for the blobstore service and all related operations.
+ */
+public class BlobstoreManager {
+  private BlobstoreService blobstoreService;
+  private BlobstoreServiceConstantFields blobstoreServiceConstantFields;
+  private BlobInfoFactory blobInfoFactory;
+
+  /**
+   * Creates a blobstore manager object by calling the blobstore service factory
+   */
+  public BlobstoreManager() throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    this.blobstoreService = blobstoreService;
+    this.blobstoreServiceConstantFields = new BlobstoreServiceConstantFields(blobstoreService);
+    this.blobInfoFactory = new BlobInfoFactory();
+  }
+
+  /**
+   * Injection code for testing.
```

Comment:
Joe mentioned recently in a PR that we shouldn't say "for testing" in javadocs for public methods. We should just like treat it like it's a part of the API

Reply:
Okay, updated

Reply:
Thanks